app/controllers/budgets_controller.rb

```diff
@@ -1,40 +1,41 @@
 class BudgetsController < ApplicationController
   before_action :set_budget, only: %i[ show edit update destroy ]
 
-  # GET /budgets or /budgets.json
+  # GET /budgets
   def index
-    @budgets = Budget.all
+    @budgets = Budget.all.order(:name)
   end
 
-  # GET /budgets/1 or /budgets/1.json
+  # GET /budgets/1
   def show
+    @budget = Budget.find(params[:id])
+    @category = Category.find(1)
```

Comment:
can remove this as its not used in show.html
and its hard coded to find category with  id = 1